File: implementations/hotchocolate/Types/ProductDimension.cs
using ApolloGraphQL.HotChocolate.Federation;

namespace Products;

[Shareable]
public class ProductDimension
{
    public ProductDimension(string size, double weight, string? unit)
    {
        Size = size;
        Weight = weight;
        Unit = unit;
    }

    public string? Size { get; }

    public double? Weight { get; }

    [Inaccessible]
    public string? Unit { get; }
}

File: implementations/smallrye-graphql/src/main/java/org/acme/microprofile/graphql/ProductDimension.java
package org.acme.microprofile.graphql;

import io.smallrye.graphql.api.federation.Shareable;
import io.smallrye.graphql.api.federation.Inaccessible;
import org.eclipse.microprofile.graphql.NonNull;

@Shareable
public class ProductDimension {
    private final String size;
    private final Float weight;
    @Inaccessible
    private final String unit;

    public ProductDimension(String size, float weight, String unit) {
        this.size = size;
        this.weight = weight;
        this.unit = unit;
    }

    public String getSize() {
        return size;
    }

    public Float getWeight() {
        return weight;
    }

    public String getUnit() {
        return unit;
    }
}
